packages/@azure/servicebus/data-plane/test/topicFilters.spec.ts

```diff
@@ -126,28 +125,16 @@ async function sendOrders(): Promise<void> {
 }
 
 async function receiveOrders(client: SubscriptionClient): Promise<ServiceBusMessage[]> {
-  let errorFromErrorHandler: Error | undefined;
   const receivedMsgs: ServiceBusMessage[] = [];
   const receiver = client.getReceiver();
-  receiver.receive(
-    (msg: ServiceBusMessage) => {
-      receivedMsgs.push(msg);
-      return Promise.resolve();
-    },
-    (err: Error) => {
-      if (err) {
-        errorFromErrorHandler = err;
-      }
+  const msgs = await receiver.receiveBatch(100);
+  if (msgs) {
+    for (let index = 0; index < msgs.length; index++) {
+      receivedMsgs.push(msgs[index]);
```

Comment:
Why push elements of `msgs` into `receivedMsgs` and then return the latter, instead of returning the former?